intellij-toml/src/main/grammars/TomlParser.bnf

```diff
@@ -42,16 +42,14 @@ private FileForm_recover ::=
   !<<atNewLine (BARE_KEY | BARE_KEY_OR_NUMBER | BARE_KEY_OR_DATE | BASIC_STRING | LITERAL_STRING | '[')>>
 private EatUntilNextLine_recover ::= !<<atNewLine <<any>>>>
 
-KeyValue ::= LongKey '=' <<atSameLine Value>> { pin = 1 }
+KeyValue ::= Key '=' <<atSameLine Value>> { pin = 1 }
 NewLineKeyValue ::= <<atNewLine KeyValue>> {
     elementType = KeyValue
     recoverWhile = EatUntilNextLine_recover
 }
 
-fake Key ::= BasicKey ('.' BasicKey)*
-LongKey ::= BasicKey ('.' BasicKey)* { elementType = Key }
-ShortKey ::= BasicKey { elementType = Key }
-private BasicKey ::= BareKey | BASIC_STRING | LITERAL_STRING
+Key ::= SimpleKey ('.' SimpleKey)*
```

Comment:
I don't like `SimpleKey` name. Maybe `KeySegment`?
Also, there is an option to parse keys with left recursion. We parser Rust paths this way. In this case, SimpleKey will not be needed. Pros/cons?

Reply:
Discussed in person. As a result:
- rename `SimpleKey` to `KeySegment`
- we currently don't know the actual cons of "parse keys with left recursion". Also, original TOML grammar parse keys also as segments list so we decided to keep current implementation